### PR TITLE
Web UI: Reset the user-entered source value after changing the problem source dropdown.

### PR DESCRIPTION
**Pull request by kwight**
#### Reference Issues/PRs
Fixes https://github.com/princeton-nlp/SWE-agent/issues/362

#### What does this implement/fix? Explain your changes.
Use `useEffect` to monitor the problem source dropdown, and when it changes, reset the user value to empty (so that the default text appears for that field).

The textarea corresponding to the "Write" dropdown doesn't need changes since the actual markup is updated on switch (between `input`s and `textarea`).

after

#### Testing Instructions
1. On a fresh page load, enter something into the "GitHub issue URL" field.
2. Switch the dropdown above it to "Local file", and verify the "Path to local .MD..." field is empty (displaying the placeholder text).
3. Enter data in that field again, and switch the dropdown back to its original value. Verify the "GitHub issue URL" field is empty again.

### CODE REVIEW

File: sweagent/frontend/src/components/controls/LRunControl.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { Tab, Tabs } from "react-bootstrap";
 import Form from "react-bootstrap/Form";
-import "../../static/runControl.css";
 import { PlayFill, StopFill } from "react-bootstrap-icons";
-
 import { Link } from "react-router-dom";
+import "../../static/runControl.css";
```
Comment (kwight) [inline sweagent/frontend/src/components/controls/LRunControl.js:6]: These changes are just removing empty lines and placing the CSS import after the components.

File: sweagent/frontend/src/components/controls/LRunControl.js [hunk 2]
```diff
@@ -19,18 +18,20 @@ function LRunControl({
 }) {
   // ps = problem statement
   const [psType, setPsType] = useState("gh");
+  const [psInputValue, setPsInputValue] = useState("");
   const defaultInstallCommand = "pip install --editable .";
-
   const defaultPS =
     "https://github.com/marshmallow-code/marshmallow/issues/1357";
 
-  const defaultRepo = "https://github.com/swe-agent-demo/marshmallow";
```
Comment (kwight) [inline sweagent/frontend/src/components/controls/LRunControl.js:27]: This variable was unused.

**Comment by kwight** [comment]
Side note, dumping this issue and the relevant file into GPT4o gave a near-perfect fix (I actually assumed refs would be needed, but this works great).

File: sweagent/frontend/src/components/controls/LRunControl.js [hunk 3]
```diff
@@ -89,23 +92,6 @@ function LRunControl({
   }
 
   function getEnvInput() {
-    // Get environment configuration input controls based on the
```
Comment (klieret) [inline sweagent/frontend/src/components/controls/LRunControl.js:92]: We will probably still want to bring this one back, so let's not remove it ;)

Comment (kwight) [reply sweagent/frontend/src/components/controls/LRunControl.js]: Oops, restored in 3b7c3a2ef339f3a1e4cbf65bd1c55d93db4092c0 👍

**Comment by ofirpress** [comment]
Cool, thanks!

**Comment by ofirpress** [comment]
@klieret can we merge this?

**Review by klieret — APPROVED** [review]
Sorry that I never got to merging this (we just got to busy with other things). Thank you for your work!

**Pull request merged**